### DISCUSSION TITLE
How to edit sentence splitter rules in medSpacy?

### DISCUSSION THREAD

**Discussion by jfredfan** [medspacy/medspacy#189] [Q&A]
Hi - 
Could you provide sample codes and instructions for adding rules of the sentence splitter in medSpacy? There seems to be example for PyRuSH itself, but it's unclear how to access/modify the rules once wrapped in medSpacy...

Thanks,
Fred

**Comment by burgersmoke** [comment]
@jianlins  Any chance you could respond to this one?  I'm not sure I fully understand the question but perhaps you do?

**Comment by OwaisAzad** [comment]
we first load the default **medspacy** (nlp = medspacy.load()) pipeline and access the **MedspacySentenceSplitter** component. Then we define our custom sentence-splitting rules as functions that take a **Doc** object and return a list of int offsets representing sentence boundaries. We also define a boolean flag indicating whether to merge the previous sentence with the current one if there's no punctuation between them.

Finally, we loop through the custom rules and add them to the **MedspacySentenceSplitter** using the **add_rule()** method. Each rule is a tuple containing the string ID for the rule, the customs rule function, and the merge flag.

Once you've added your custom rules, you can use the modified **medspacy** pipeline to process your documents and split them into sentences according to your custom rules.
